smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -175,66 +185,97 @@ impl<'a, R: RngCore, BackingStore: trie::BackingStoreLoad> machine::Host<Artifac
+                    filename,
+                    line,
+                    column,
+                    msg,
+                    remaining_energy: 0.into(), // debug host does not have energy.
+                });
+            }
+            "state_lookup_entry" => state_lookup_entry(memory, stack, energy, state)?,
+            "state_create_entry" => state_create_entry(memory, stack, energy, state)?,
+            "state_delete_entry" => state_delete_entry(memory, stack, energy, state)?,
+            "state_delete_prefix" => state_delete_prefix(memory, stack, energy, state)?,
+            "state_iterate_prefix" => state_iterator(memory, stack, energy, state)?,
+            "state_iterator_next" => state_iterator_next(stack, energy, state)?,
+            "state_iterator_delete" => state_iterator_delete(stack, energy, state)?,
+            "state_iterator_key_size" => state_iterator_key_size(stack, energy, state)?,
+            "state_iterator_key_read" => state_iterator_key_read(memory, stack, energy, state)?,
+            "state_entry_read" => state_entry_read(memory, stack, energy, state)?,
+            "state_entry_write" => state_entry_write(memory, stack, energy, state)?,
+            "state_entry_size" => state_entry_size(stack, energy, state)?,
+            "state_entry_resize" => state_entry_resize(stack, energy, state)?,
+            "set_slot_time" => {
+                let slot_time = unsafe { stack.pop_u64() };
+                self.slot_time = Some(slot_time);
+            }
+            "get_slot_time" => {
+                let slot_time = self.slot_time.context("slot_time is not set")?;
+                stack.push_value(slot_time);
+            }
+            "set_receive_self_address" => {
+                let addr_ptr = unsafe { stack.pop_u32() };
+                let mut cursor = Cursor::new(memory);
+
+                cursor
+                    .seek(SeekFrom::Start(addr_ptr))
+                    .map_err(|_| anyhow!("unable to read bytes at the given position"))?;
+
+                self.address = Some(ContractAddress::deserial(&mut cursor)?);
+            }
+            "get_receive_self_address" => {
+                let addr_ptr = unsafe { stack.pop_u32() };
+                let mut cursor = Cursor::new(memory);
+
+                cursor
+                    .seek(SeekFrom::Start(addr_ptr))
+                    .map_err(|_| anyhow!("unable to read bytes at the given position"))?;
+
+                self.address
+                    .context("self_address is not set")?
+                    .serial(&mut cursor)
+                    .map_err(|_| anyhow!("unable to serialize the self address"))?;
+            }
+            "set_receive_self_balance" => {
+                let balance = unsafe { stack.pop_u64() };
+                self.balance = Some(balance);
+            }
+            "get_receive_self_balance" => {
+                let balance = self.balance.context("no balance was set")?;
```

Comment:
Maybe this should just be set to zero instead of being an option? Or is it better to be more explicit?

Reply:
Either is fine with me, the old testing infrastructure actually used zero, but I personally lean towards having the user be explicit, since the context of the unit test becomes more explicit.